File: Day8/Solution.java
class Solution{
    public static void main(String []argh){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        Map<String, Integer> phoneBook = new HashMap<String, Integer>();
        for(int i = 0; i < n; i++){
            String name = in.next();
            int phone = in.nextInt();
            phoneBook.put(name, phone);

        }
        while(in.hasNext()){
            String s = in.next();
            if(phoneBook.containsKey(s))
                System.out.println(s + "=" + phoneBook.get(s));
            else
                System.out.println("Not found");


        }
        in.close();
    }
}

File: Day8/Solution.rs
use std::collections::HashMap;
use std::io;
use std::io::BufRead;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut phonebook = HashMap::new();
    let mut n_numbers = String::new();

    io::stdin().read_line(&mut n_numbers)?;
    let n: i32 = n_numbers.trim().parse()?;
    for _ in 0..n {
        let mut line = String::new();
        io::stdin().read_line(&mut line)?;
        let mut words = line.split_whitespace();
        let name = words.next().unwrap().to_string();
        let number = words.next().unwrap().to_string();
        phonebook.insert(name.clone(), number.clone());
    }
    for query in io::stdin().lock().lines() {
        let name = query.unwrap();
        let result = match phonebook.get(&name) {
            Some(number) => format!("{}={}", name, number),
            None => "Not found".to_string(),
        };
        println!("{}", result);
    }
    Ok(())
}
